apps/mobile/src/features/balances/stacks/use-get-stacks-addresses.ts

```diff
@@ -8,8 +8,14 @@ export function useGetStacksAddresses(accountId?: AccountId) {
     return signers.list.map(signer => signer.address);
   }
 
-  const { fingerprint } = accountId;
-  return signers.list
-    .filter(signer => signer.descriptor.includes(fingerprint))
-    .map(signer => signer.address);
+  const { accountIndex } = accountId;
+  console.log('signers.list', signers.list);
+  // is this wrong? all signer descriptors underneath the same wallet have the same fingerprint
```

Comment:
I was naively using `fingerprint` to get stacks addresses to use to check balance. 

That was returning all addresses associated with that fingerprint. 

I think I need to change this to use the fingerprint in combination with `accountIndex` but account index is not incrementing in signers.

The code I added here - `signers.list[accountIndex]` loads the correct balances but by coincidence of the order of the signers array. 

 I think the code should be more like:
 ```
return signers.list
    .filter(signer => signer.descriptor.includes(fingerprint) && signer.accountIndex === accountIndex)
    .map(signer => signer.address);
```